Lib/email/utils.py

```diff
@@ -212,9 +242,18 @@ def parseaddr(addr):
     Return a tuple of realname and email address, unless the parse fails, in
     which case return a 2-tuple of ('', '').
     """
-    addrs = _AddressList(addr).addresslist
-    if not addrs:
-        return '', ''
+    if type(addr) == list:
+        addr = addr[0]
+
+    if type(addr) != str:
+        return ('', '')
```

Comment:
We can avoid using `type()` to compare the types of variables, using `isinstance()` is a better choice.
```suggestion
    if isinstance(addr, list):
        addr = addr[0]

    if not isinstance(addr, str):
        return ('', '')
```